### PR TITLE
add aws manager blueprint

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: ec2/inputs.yaml.template [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+use_existing_agent_keypair: false
+use_existing_manager_group: false
+use_existing_agent_group: false
+manager_keypair_name: ec2-manager-kp
+agent_keypair_name: ec2-agent-kp
+manager_key_pair_file_path: ~/.ssh/ec2-manager-kp.pem
+agent_key_pair_file_path: ~/.ssh/ec2-agent-kp.pem
+mananger_security_group_name: ec2-manager-security-group
+agent_security_group_name: ec2-agent-security-group
+manager_server_name: ec2-manager-server
```
Comment (idanmo) [inline ec2/inputs.yaml.template]: default values should probably be in the following format: "cloudify-*" without an ec2 prefix.

File: ec2/scripts/configure.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
```
Comment (idanmo) [inline ec2/scripts/configure.py]: 2015

File: ec2/scripts/configure.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    _upload_credentials(config_path)
+    _set_provider_config(agents_security_group, agents_keypair)
+
+
+def _upload_credentials(config_path):
+    temp = configure.BotoConfig().get_temp_file()
+    fabric.api.put(temp, config_path)
+
+
+def _set_provider_config(agents_security_group, agents_keypair):
```
Comment (idanmo) [inline ec2/scripts/configure.py]: What about provider context? shouldn't it be set as a runtime property of the manager node?

File: ec2/inputs.yaml.template (hunk 1 above)
Comment (EarthmanT) [inline ec2/inputs.yaml.template]: okay

File: ec2/scripts/configure.py (hunk 2 above)
Comment (EarthmanT) [inline ec2/scripts/configure.py]: okie dokie

File: ec2/scripts/configure.py (hunk 3 above)
Comment (EarthmanT) [inline ec2/scripts/configure.py]: okay, I added that but who uses it?

File: aws/aws-manager-blueprint.yaml [hunk 4]
```diff
@@ -0,0 +1,243 @@
+# DSL version, should appear in the main blueprint.yaml
+# and may appear in other imports. In such case, the versions must match
```
Comment (idanmo) [inline aws/aws-manager-blueprint.yaml]: Please remove these comments...

File: aws/aws-manager-blueprint.yaml [hunk 5]
```diff
@@ -0,0 +1,265 @@
+tosca_definitions_version: cloudify_dsl_1_1
```
Comment (idanmo) [inline aws/aws-manager-blueprint.yaml]: Perhaps the file name should be aws-ec2-manager-blueprint.yaml

File: aws/aws-manager-blueprint.yaml [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+      private_key_path: { get_input: agent_key_pair_file_path }
+      aws_config: { get_property: [ aws_configuration, aws_config ] }
+
+  management_security_group:
+    type: cloudify.aws.nodes.SecurityGroup
+    properties:
+      resource_id: { get_input: mananger_security_group_name }
+      use_external_resource: { get_input: use_existing_manager_group }
+      description: Security group for Cloudify Manager VM
+      rules:
```
Comment (ran-z) [inline aws/aws-manager-blueprint.yaml]: probably want to switch up the security-groups inter-dependency

File: aws/inputs.yaml.template [hunk 7]
```diff
@@ -0,0 +1,25 @@
+aws_access_key_id:
```
Comment (ran-z) [inline aws/inputs.yaml.template]: separate mandatory and optional sections

File: aws/scripts/configure.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+                'id': run_props[constants.EXTERNAL_RESOURCE_ID],
+            }
+
+    provider = {
+        'resources': resources
+    }
+
+    ctx.instance.runtime_properties['provider_context'] = provider
+
+    temp_config = tempfile.mktemp()
```
Comment (ran-z) [inline aws/scripts/configure.py]: remove

File: aws/aws-manager-blueprint.yaml (hunk 5 above)
Comment (EarthmanT) [inline aws/aws-manager-blueprint.yaml]: changed

File: aws/aws-manager-blueprint.yaml (hunk 6 above)
Comment (EarthmanT) [inline aws/aws-manager-blueprint.yaml]: switched

File: aws/scripts/configure.py (hunk 8 above)
Comment (EarthmanT) [inline aws/scripts/configure.py]: did

File: aws-ec2/scripts/configure.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+# Cloudify Imports
+from cloudify import ctx
+from ec2 import configure
+from ec2 import constants
+
+
+def configure_manager(
+        manager_config_path=constants.AWS_DEFAULT_CONFIG_PATH,
+        aws_config={}):
```
Comment (dankilman) [inline aws-ec2/scripts/configure.py]:
```
aws_config=None)
awc_config = aws_config or {}
```

**Pull request merged**